src/amo/components/AddonReviewCard/index.js

```diff
@@ -481,7 +486,7 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
           </a>
         ) : null}
 
-        {flaggable && review ? (
+        {flaggable && review && showFlagButton ? (
```

Comment:
You also need to add a test that makes sure the flag button is not hidden when the user is signed out. After writing the test, you can comment out the part that allowed non-signed in users to see the flag link; the test should fail. This helps you know that the test is working.